src/string_enums/base_enum.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+	Copyright (C) 2008 - 2021
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+namespace string_enums
+{
+/**
+ * The base template for associating string values with enum values.
+ * Implementing classes should not set custom int values for their enum.
+ * The number of enum values must match the number of elements in the @a values_ array.
+ */
+template<class T>
+struct enum_base
```

Comment:
Please swap the order of the words either in this line, or in the filename; otherwise we have both `enum_base` and `base_enum.hpp`.

Reply:
Done.